core/util.go

```diff
@@ -364,11 +364,13 @@ func GetBuildHost() (retID string) {
 	return
 }
 
-// UniqueNames returns the set of all unique names in the input.
+// UniqueNames returns the set of all unique names in the input after. The
+// returned names will all be lowercased and will have been checked against each
+// other in their lowercase form..
 func UniqueNames(names []string) (unique []string) {
```

Comment:
Okay, it looks like the only other place calling UniqueNames is ra.MatchesCSR which is not being used anywhere.

So, UniqueNames doesn't need this feature if we delete MatchesCSR. Meanwhile, the CA code with the ToLower is still present and okay.